components/styled/input/src/atoms/text.tsx

```diff
@@ -43,11 +41,19 @@ const StyledSearchIcon = styled(SearchIcon)`
   }
 `
 
-const TextInput: FC<InputHTMLAttributes<HTMLInputElement>> = ({ placeholder, ...rest }) => (
+const TextInput: FC<ITextInputProps> = ({ placeholder, icon, ...rest }) => (
   <TextInputContainer>
-    <StyledSearchIcon size="medium" />
-    <StyledTextInput type="text" placeholder={placeholder} {...rest} />
+    {icon && <StyledIconWrapper>{icon}</StyledIconWrapper>}
+    <StyledTextInput type="text" placeholder={placeholder} {...rest} hasIcon={icon !== undefined} />
   </TextInputContainer>
 )
 
+interface ITextInputProps extends InputHTMLAttributes<HTMLInputElement> {
+  icon?: ReactElement
```

Comment:
You could make this an `Iconprops` from `@ltht-react/types` then render the `Icon` component accordingly, then you could revert the `styled.div` back to a `styled(Icon)`.

Reply:
Forgive me, from `@ltht-react/icon`. exported from icon.tsx and used in IconButton, for example

Reply:
Lovely idea, that component saves so much faffing. Done.